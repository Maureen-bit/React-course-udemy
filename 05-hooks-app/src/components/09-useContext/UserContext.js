import { createContext } from 'react'

/* un context es un order higher component ya que va a contener m√°s componentes adentro */
export const UserContext = createContext(null);
