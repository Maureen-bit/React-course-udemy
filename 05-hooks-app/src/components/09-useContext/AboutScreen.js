import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);
  const handleClick = () => {
    /* Cuando el context sufre una modificaci√≥n notifica a todos los hijos sobre el cambio y hace que se 
    redibuje cada una de las partes afectadas */
    setUser({})
  }
  return (
    <div>
        <h1>AboutScreen</h1>
        <hr/>
        <pre>
            {JSON.stringify(user, null, 3)}
        </pre>
        <button className='btn btn-warning' 
        onClick={handleClick}
        > 
          Logout 
        </button>
    </div>
  )
}
