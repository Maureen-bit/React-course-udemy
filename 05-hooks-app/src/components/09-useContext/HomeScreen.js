import React, {useContext} from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {
  /* Obtener el valor de nuestro useContext creado para guardar datos del usuario */
  const { user } = useContext(UserContext);
  return (
    <div>
        <h1>HomeScreen</h1>
        <hr/>
        <pre>
            {JSON.stringify(user, null, 3)}
        </pre>
    </div>
  )
}
