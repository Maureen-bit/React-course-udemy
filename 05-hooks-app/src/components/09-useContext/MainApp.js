import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainApp = () => {
    /* UserContext: va a envolver todos los componentes que encuentre en AppRouter, es decir que, la información
    que se guarde en este context estará disponible para todos estos hijos.
    Lo que se encuentra en la propiedad value es lo que voy a compartir  */
    const [user, setUser] = useState({})

  return (
    <UserContext.Provider value={{user, setUser}}>
        <AppRouter />
    </UserContext.Provider>
  )
}
