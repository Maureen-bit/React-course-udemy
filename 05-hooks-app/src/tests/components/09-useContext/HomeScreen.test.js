import React from 'react';
import { mount, shallow } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Tests in <HomeScreen /> Component', () => { 
    const user = {
        name: 'Maureen',
        email: 'maureen@gmail.com'
    }
    const wrapper = mount(
        /* Provider: provee el contexto que se creo para que la prueba se ejecute correctamente */
        <UserContext.Provider value={{user}}>
            <HomeScreen />
        </UserContext.Provider>
        );
    test('It should renders correctly', () => { 
        expect(wrapper).toMatchSnapshot();
     });
 })