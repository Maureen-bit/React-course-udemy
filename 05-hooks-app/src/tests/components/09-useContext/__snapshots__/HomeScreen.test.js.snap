// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <HomeScreen /> Component It should renders correctly 1`] = `<HomeScreen />`;
